torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1868,6 +1875,13 @@ def _simplify_floor_div(self, expr):
             self.evaluate_expr(eq_expr)
         return self.simplify(expr)
 
+    def _add_guard(self, expr: "sympy.Expr") -> None:
+        stack = ''.join(traceback.format_list(traceback.extract_stack()[:-1]))
+        guard = ShapeGuard(expr, stack)
+        if torch._dynamo.config.print_shape_guards:
+            torch._dynamo.guards.log.warning(f"Adding shape guard {expr}:\n{stack}")
```

Comment:
We already have guard printing right? Why do we want more?

Reply:
You mean at the end, right? @ezyang mentioned it would have been useful for him to see the stack trace as they are added when debugging (e.g.) https://github.com/pytorch/pytorch/issues/93468#issuecomment-1412520722

Yes, I realize it is also added to the shape guard. So we could also print guards at the end (by default we omit the stacks, I suppose that can be changed under a flag...which one?).

Reply:
That makes sense, I would reuse the flag for dynamo guard printing then, and add a comment to the redundancy.